composer.json

```diff
@@ -4,7 +4,8 @@
     "type": "library",
     "require": {
         "php": ">=7.1.0",
-        "psr/cache": "^1.0"
+        "psr/cache": "^1.0",
+        "symfony/cache": "^4.3"
```

Comment:
Can you pin this to this version? Does that break blogs, or other consumers?

Reply:
I'm not too sure if adding "symfony/cache" was a good idea.
This could cause issues with different versions so I was wondering, what if instead of using `return new CacheItem();` I pass an instance of an empty item within the DI? I think it would be a better approach, what do you think?

Reply:
Would that work though? The Symfony DI layer only creates one instance of the class and it's used everywhere. I suspect that might lead to the same cacheitem being reused in different places, which could cause weird bugs. Has the API for symfony cache changed between versions? Version could just be >2.0 or something otherwise, no?

Reply:
Let me show you an example of the constructor
```
    public function __construct(
        LoggerInterface $logger,
        CacheItemPoolInterface $cachePool,
        string $prefix,
        int $resilienceTtl,
        CacheItemInterface $emptyResponse,
        array $cacheTimes = [],
        array $whitelistedExceptions = []
    ) {
        $this->logger = $logger;
        $this->resilienceTtl = $resilienceTtl;
        $this->emptyResponse = $emptyResponse;
        $this->whitelistedExceptions = $whitelistedExceptions;
        parent::__construct($cachePool, $prefix, $cacheTimes);
    }
```

and an example of the services.yaml
```
    cacheitem: { class: Symfony\Component\Cache\CacheItem }

    BBC\ProgrammesCachingLibrary\CacheWithResilience:
        arguments: ['@logger', '@cache.programmes', 'programmes-frontend.%cosmos_component_release%', 1000, '@cacheitem', [], ['Doctrine\DBAL\Exception\ConnectionException', 'Doctrine\DBAL\Exception\ServerException']]

```

I'm not sure if I understand what could be wrong about using the instance of this class,  the calss is mean to be allways an empty Item so we can return this in line 66. I have tested it and seems working fine.
This way we shouldn't have any problem with symfony versions

I haven't investigate if there are any changes in the symfony version but I have the feeling that composer could complain at certain point if blogs and frontend have different required versions, not sure if worth taking the risk, feels like the wrong approach but maybe is just me

Reply:
@Tasiobg the problem with passing in the cache item is that everything using the cache class will get the same one. 
Consider the following code in frontend: https://github.com/bbc/programmes-frontend/blob/develop/src/ExternalApi/Client/HttpApiMultiClient.php#L119 
Here it asks the cache for a specific key. Assume that the key is in the cache, but stale, and therefore the single CacheItem class is returned.

Then look at the set in that class: https://github.com/bbc/programmes-frontend/blob/develop/src/ExternalApi/Client/HttpApiMultiClient.php#L161 . It's using the same CacheItem returned from the DI, setting stuff in it and storing it. 

The next thing that calls get() on a key that's in the cache but stale will get the CacheItem that the HttpApiMultiClient class just set. Resulting in weird bugs. 

I could be wrong, but I can't see how that can work reliably. If you want to go with this approach to avoid requiring symfony/cache, you would need to pass in a CacheItemFactory or a generator function and use that I think.